private/buf/cmd/buf/command/format/format.go

```diff
@@ -0,0 +1,536 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package format
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/buf/buffetch"
+	"github.com/bufbuild/buf/private/buf/bufformat"
+	"github.com/bufbuild/buf/private/buf/bufwork"
+	"github.com/bufbuild/buf/private/bufpkg/bufanalysis"
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/command"
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"github.com/bufbuild/buf/private/pkg/storage/storagemem"
+	"github.com/bufbuild/buf/private/pkg/storage/storageos"
+	"github.com/bufbuild/buf/private/pkg/stringutil"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.uber.org/multierr"
+)
+
+const (
+	configFlagName          = "config"
+	diffFlagName            = "diff"
+	diffFlagShortName       = "d"
+	disableSymlinksFlagName = "disable-symlinks"
+	errorFormatFlagName     = "error-format"
+	excludePathsFlagName    = "exclude-path"
+	outputFlagName          = "output"
+	outputFlagShortName     = "o"
+	pathsFlagName           = "path"
+	writeFlagName           = "write"
+	writeFlagShortName      = "w"
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appflag.Builder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <input>",
+		Short: "Format all Protobuf files from the specified input and output the result.",
+		Long: `
+By default, the formatted content is written to stdout. For example,
+
+# Write the formatted file to stdout
+$ buf format simple/simple.proto
+syntax = "proto3";
+
+package simple;
+
+message Object {
+  string key = 1;
+  bytes value = 2;
+}
+
+# Write the formatted directory to stdout
+$ buf format simple
+syntax = "proto3";
+
+package simple;
+
+message Object {
+  string key = 1;
+  bytes value = 2;
+}
+
+Write the result to a specified output file or directory with -o. For example,
+
+# Write the formatted file to another file
```

Comment:
Can the first example be what happens be `buf format` and then `buf format -w`, explaining that this will use `.`, ie whatever module or workspace is in the current directory? It might also just be a note that "most people will want to do `buf format -w`".

Reply:
Yea I agree - I'll rearrange this a bit.